src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +109,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (String origin : StructureLoadingUtils.getOriginMods().keySet())
             {
-                section = hut;
-            }
-            else
-            {
-                hut = "";
+                if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(origin, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+                {
+                    section = hut;
+                    break;
+                }
+                else
```

Comment:
This is not a correct logic rewrite btw, if you iterate first then you end up with hut = ""

Reply:
... Did I seriously.... damn.

Reply:
Alright, I just removed the else block. From what I can see, it isn't necessary anyway.

Reply:
Even now it's not correct :D you are missing the else branch logic now

Reply:
And also the list is probably needed? Depends on usage in mcol

Reply:
.... Wow. I'm just gonna go sit in a corner now.

Reply:
@Nightenom, the HUTS list is not referenced from minecolonies. I'm pretty sure it was originally populated so that it could be checked against by the BuildTool. I went through each time StructureName is imported by mcol, it's never used.